tests/unit/test_logging.py

```diff
@@ -0,0 +1,273 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Tests for the log_tools module."""
+
+import json
+import logging
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+
+import pytest
+
+from hexkit.log_tools import (
+    JsonFormatter,
+    LoggerFactory,
+    LoggingConfig,
+    RecordCompiler,
+    StructuredLogger,
+)
+
+VALID_SERVICE_NAME = "test"
+VALID_INSTANCE_ID = "001"
+VALID_CONFIG = LoggingConfig(
+    log_level="INFO",
+    service_name="hexkit-test",
+    service_instance_id="0",
+)
+
+DEFAULT_KEYS = (  # all the keys present in the resulting log
+    "timestamp",
+    "service",
+    "instance",
+    "level",
+    "name",
+    "correlation_id",
+    "message",
+    "details",
+)
+
+ADDED_KEYS = (  # a list of non-standard keys added by the adapter/record compiler
+    "timestamp",
+    "service",
+    "instance",
+    "level",
+    "correlation_id",
+    "details",
+)
+
+DEFAULT_CONFIG = LoggingConfig(service_name="", service_instance_id="")
+
+
+@pytest.fixture(autouse=True)
+def reset_config():
+    """Reset the config before and after each test."""
+    LoggerFactory.configure(log_config=DEFAULT_CONFIG)
+    yield
+    LoggerFactory.configure(log_config=DEFAULT_CONFIG)
```

Comment:
If this is used by all tests, it should probably suffice to just configure once in the beginning